### ISSUE TITLE
how the model ensure semantic information goes to z rather than x_t?

### ISSUE THREAD

**Issue by betterze** [phizaz/diffae#30]
Dear diffae team, 

Thank you for sharing this great work, I really enjoy it. 

I paly with the model a bit. When I fix z and randomly sample x_t, the output images are almost the same, with some small variaces. how the model ensure semantic information goes to z rather than x_t (starting gaussian noise)? Maybe because the model is trained with reconstruction loss, z is the only source for the target image information, and x_t is randomly sampled even for the same target image. So the model learn to 'ingore' x_t and therefore x_t only contains little semantic information. 

In this [repo](https://github.com/justinpinkney/stable-diffusion), Justin takes the Unet from stable diffusion and the image encoder from CLIP, and train the model using image reconstruction loss. If my intuition in last paragraph is correct, then given the same image as input, we randomly sample x_t, the output images will be almost the same. But this not the [case](https://twitter.com/Buntworthy/status/1565704770056294400) in Justin's model, changing x_t causes a large change (including semantic change) in output image. This mean the x_t in Justin's model contain more semantic information than the x_t in DA model. Could you tell me why this is happening? 

Thank you for your help.

Best Wishes,

Zongze

**Comment by phizaz** [comment]
This is the question we have been pondering for so long as well. I don't pretend to have the full answer. The following is the best intuition I can give you.

DiffAE doesn't learn a "semantic" latent code. It learns a latent code that packs "the most bang for the buck" under the implicit bias of the decoder, in this case, a diffusion model. 


There is no reason to believe that a semantic latent code, corresponding to human judgment, is the same as one that packs the most bang for the buck, due to the strong emphasis of the reconstruction loss. 
The fact that we observed DiffAE learn a semantically meaningful latent code should be considered **accidental**. Maybe, due to the dataset we used, FFHQ256, which is heavily cropped and properly aligned. We should not expect this quality of DiffAE to transfer to other datasets in general.

What to do if I want to control what's kept in the latent code?
There are two ways I could think of: change the loss function to something more semantic, or using a different **encoder**.

Using a different loss function resulting in a different latent code is easy to understand, but finding a substitute for the reconstruction loss is hard to do.

A more doable way is through the encoder. Let's consider a CLIP encoder. It was trained with correspondence between images and texts. It tends to capture high-level abstract concepts due to the text information given to describe an image. The latent code from this CLIP encoder is expected to capture high-level concepts rather than precise appearances. 

One could extend this intuition to another encoder as ArcFace. Since ArcFace is trained to be invariant to most things expect the face identity. The latent code from this encoder should be deprived of most information except identity, i.e., the decoded face may not share the same pose an the original face.
